src/server/services/wms/qgswmsgetserversettings.cpp

```diff
@@ -0,0 +1,157 @@
+    aboutServer["Version"] = caption;
+    if ( QString( Qgis::QGIS_DEV_VERSION ) == QLatin1String( "exported" ) )
+    {
+      caption = QStringLiteral( "https://github.com/qgis/QGIS/tree/release-%1_%2" )
+                .arg( Qgis::QGIS_VERSION_INT / 10000 ).arg( Qgis::QGIS_VERSION_INT / 100 % 100 );
+      aboutServer["QGIS code branch"] = caption;
+    }
+    else
+    {
+      caption = QStringLiteral( "https://github.com/qgis/QGIS/commit/%2" ).arg( Qgis::QGIS_DEV_VERSION );
+      aboutServer["QGIS code revision"] = caption;
+    }
+    aboutServer["Compiled against Qt"] = QT_VERSION_STR;
+    aboutServer["Running against Qt"] = qVersion();
+
+    aboutServer["Compiled against GDAL/OGR"] = GDAL_RELEASE_NAME;
+    aboutServer["Running against GDAL/OGR"] = GDALVersionInfo( "RELEASE_NAME" );
+
+    aboutServer["Compiled against GEOS"] = GEOS_CAPI_VERSION;
+    aboutServer["Running against GEOS"] = GEOSversion();
+
+    aboutServer["Compiled against GEOS"] = SQLITE_VERSION;
+    aboutServer["Running against GEOS"] = sqlite3_libversion();
+
+#ifdef HAVE_POSTGRESQL
+    aboutServer["PostgreSQL Client Version"] = QStringLiteral( PG_VERSION );
+#else
+    aboutServer["PostgreSQL Client Version"] = QStringLiteral( "No support" );
+#endif
+
+    aboutServer["SpatiaLite Version"] = spatialite_version();
+    aboutServer["QWT Version"] = QWT_VERSION_STR;
+    aboutServer["QScintilla2 Version"] = QSCINTILLA_VERSION_STR;
+
+#if PROJ_VERSION_MAJOR > 4
+    PJ_INFO info = proj_info();
+    aboutServer["Compiled against PROJ"] = QStringLiteral( "%1.%2.%3" ).arg( PROJ_VERSION_MAJOR ).arg( PROJ_VERSION_MINOR ).arg( PROJ_VERSION_PATCH );
+    aboutServer["Running against PROJ"] = info.release;
+#else
+    aboutServer["PROJ.4 Version"] = QString::number( PJ_VERSION );
+#endif
+
+    aboutServer["OS Version"] = QSysInfo::prettyProductName();
+    aboutServer["Kernel version"] = QSysInfo::kernelVersion();
+    aboutServer["Hostname"] = QSysInfo::machineHostName();
+    aboutServer["CPU Cores"] = QThread::idealThreadCount();
+
+#ifdef Q_OS_LINUX
+    long pages = sysconf( _SC_PHYS_PAGES );
+    long pagesAvailable = sysconf( _SC_AVPHYS_PAGES );
+    long page_size = sysconf( _SC_PAGE_SIZE );
+    aboutServer["System Memory (Mb)"] = QStringLiteral( "%1" ).arg( pages * page_size / 1024 / 1024 );
+    aboutServer["Available memory (Mb)"] = QStringLiteral( "%1" ).arg( pagesAvailable * page_size / 1024 / 1024 );
+#endif
+
+    return aboutServer;
+  }
+
+  void writeServerSettings( QgsServerInterface *serverIface, QgsServerResponse &response )
+  {
+    QJsonObject serverSettings;
+    serverSettings["about"] = about();
+    serverSettings["settings"] = serverIface->serverSettings()->logSummaryJson();
+    serverSettings["fonts"] = availableFonts();
+
+    QJsonObject serverPlugins;
+    QJsonArray availablePlugins = {};
+    QStringList plugins = QgsServerPlugins::serverPlugins();
+    for ( int i = 0; i < plugins.size(); ++i )
```

Comment:
Do you think we could do something like `for ( const QString &plugin : plugins )` instead?

Reply:
And in this case don't foget to `const QStringList plugins ...` on 146